### PR TITLE
1) Implement favourite movies feature

### PR DESCRIPTION
**Pull request by ecustovic**

### CODE REVIEW

File: app/controllers/favourites_controller.rb [hunk 1]
```diff
@@ -1,18 +1,30 @@
 class FavouritesController < ApplicationController
   before_action :require_signin
-  before_action :set_movie
+
+  def index 
+    @fav_movies = true
```
Comment (dixpac) [inline app/controllers/favourites_controller.rb]: For what is this ivar `@fav_movies` used?

File: app/views/movies/index.html.erb [hunk 2]
```diff
@@ -12,9 +13,15 @@
         <%= total_gross(movie) %>
       </h3>
 
-      <span class="stars">
-        <%= render "shared/stars", percent: movie.average_stars_as_percent %>
-      </span>
+      <%if @fav_movies %>
```
Comment (dixpac) [inline app/views/movies/index.html.erb]: `@fav_movies` is a leftover from the old implementation and not necessary

File: test/integration/favourite_flow_test.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'test_helper'
+
+class FavouriteFlowTest < ActionDispatch::IntegrationTest
+  test "user can see the list of fav movies" do 
+    sign_in users(:dino)
+    get "/movies/filter/favourites"
```
Comment (dixpac) [inline test/integration/favourite_flow_test.rb]: You need to change the tests. to point to `get favouties_url`

File: app/views/movies/index.html.erb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <% @movies.each do |movie| %>
   <section class="movie">
     <div class="image">
+
```
Comment (dixpac) [inline app/views/movies/index.html.erb]: ✂️ new line

File: app/views/favourites/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<% @movies.each do |movie| %>
+  <section class="movie">
+    <div class="image">
+
```
Comment (dixpac) [inline app/views/favourites/index.html.erb]: ✂️ new line

File: app/helpers/favourites_helper.rb [hunk 6]
```diff
@@ -1,9 +1,13 @@
 module FavouritesHelper
```
Comment (dixpac) [inline app/helpers/favourites_helper.rb]: It is better to use `link_to` here instead of the `button_to`, you can style link to look lite a button with bootstrap calsses

File: app/controllers/favourites_controller.rb [hunk 7]
```diff
@@ -1,18 +1,30 @@
...
-    favourite.destroy
-
-    redirect_to movie
+    favourite = Favourite.find_by(movie_id: params[:id], user_id: current_user.id)
+    if favourite.destroy
+      flash[:notice] = "Movie unfavorited."
+    else
+      flash[:alert] = "Unfavoriting failed."
+    end
+      redirect_to root_path
```
Comment (dixpac) [inline app/controllers/favourites_controller.rb]: Indentation is wrong here

File: app/controllers/favourites_controller.rb [hunk 8]
```diff
@@ -1,18 +1,30 @@
...
+    @movies = current_user.favourite_movies.page(params["page"])
+  end
 
   def create
-    @movie.favourites.create!(user: current_user)
-  
-    redirect_to @movie
+    favourite = Favourite.new(movie_id: params[:movie_id], user_id: current_user.id)
+    
+    if favourite.save
```
Comment (dixpac) [inline app/controllers/favourites_controller.rb]: Use a default Rails way to respond from the controller
```ruby
respond_to do |format|
  if favoutite.save
      format.html { redirect_to favourites_path, notice: "Saved as favorite!" }
  else
      format.html { redirect_to favourites_path, alert: "Favorite failed to save." }
  end
end
```

File: app/controllers/favourites_controller.rb [hunk 9]
```diff
@@ -1,18 +1,30 @@
...
+      redirect_to root_path
   end
 
   def destroy
-    favourite = current_user.favourites.find(params[:id])
-    favourite.destroy
-
-    redirect_to movie
+    favourite = Favourite.find_by(movie_id: params[:id], user_id: current_user.id)
+    if favourite.destroy
```
Comment (dixpac) [inline app/controllers/favourites_controller.rb]: No need for this check beacuase `destroy` will always succede in your use-case

File: app/controllers/favourites_controller.rb [hunk 10]
```diff
@@ -1,18 +1,28 @@
...
+      if favourite.save
+        format.html {redirect_to favourites_path, notice: "Saved as favorite!"}
+      else
+        format.html {redirect_to favourites_path, notice: "Favorite failed to save."}
+      end
+    end
   end
 
   def destroy
-    favourite = current_user.favourites.find(params[:id])
```
Comment (dixpac) [inline app/controllers/favourites_controller.rb]: Why did you change this line?

Comment (ecustovic) [reply app/controllers/favourites_controller.rb]: Because I should to implement unfave feature on Favourites list and there I could delete movie only if i pass movie_id and user_id params
